chia/cmds/coin_funcs.py

```diff
@@ -207,8 +207,14 @@ async def async_split(args: Dict[str, Any], wallet_client: WalletRpcClient, fing
         print("Try using a smaller fee or amount.")
         return
     additions: List[Dict[str, Union[uint64, bytes32]]] = []
+    puzzle_hashes: Set[bytes32] = set()  # we use a set for speed.
     for i in range(number_of_coins):  # for readability.
         target_ph: bytes32 = decode_puzzle_hash(await wallet_client.get_next_address(str(wallet_id), unique_addresses))
+        if target_ph in puzzle_hashes:  # if this ph is used by another coin that is being created, we cant use it.
+            target_ph = decode_puzzle_hash(await wallet_client.get_next_address(str(wallet_id), new_address=False))
+            if target_ph in puzzle_hashes:  # we need to get a new address because the second address is also used.
+                target_ph = decode_puzzle_hash(await wallet_client.get_next_address(str(wallet_id), new_address=True))
+        puzzle_hashes.add(target_ph)  # add ph to list of ph's that we are using to avoid duplicates.
```

Comment:
That kinda defeats the point, we want all puzzle hashes to be added, no matter if it was originally a duplicate or not.
Even if we make a new address one loop, the next loop might give us the same address and then we would have it twice. Eventually i will add a new rpc, but i wanted to keep these changes as light as possible.

Reply:
This just puts it in the `if`.  Running it outside the `if` only does something if it went inside the `if` anyways.  If it didn't go into the `if` then it is already in the set and doesn't need to be added.

Reply:
We want all puzzle hashes to be in the set, not just ones we just generated. If we put it in the if, only the ph's we just generated will be in it and it will be useless.

Reply:
I had myself backwards on this...  `:[`  sorry.